moxiworks/widget-factory
# Add expanded widget; Add error and spinner partials

## Description

In preparation for the RISMedia widget, this factors out an `external_widget_component` that will be used by both the ListTrac and the RISMedia widgets.

I've also factored out "spinner" and "error" partials and added them to the inline widget component.  In order for the "error" partial to only be shown when a critical error occurs (i.e. `@error.present?`), I also had to change the Hurdlr widget to treat "subscription not found" errors from the API as a non-critical error.

Finally, I realized that when I resized the ListTrac logo for the inline widget component, that I made it smaller than the expanded modal logo.  I've resized the logo so it should scale nicely for both use cases, and I added a migration to reattach the file.


## Validation Steps
Verify that the ListTrac widget still functions as expected.
Verify that the Hurdlr widget still functions as expected. The sign-up button should be shown for our test users that do not have valid MLS numbers.
